src/download/tests/support/mod.rs

```diff
@@ -23,6 +22,7 @@ pub fn file_contents(path: &Path) -> String {
     result
 }
 
+#[allow(dead_code)]
```

Comment:
This seems like an unusual `allow()` to have here, under what circumstances is it needed?

Reply:
`cargo test` warns I don't use this function in my test files.

Reply:
If you don't use it, should we remove it rather than leave it there?  Or is there perhaps one more test which could be added to make it not-dead code?

Reply:
This support file is used by both my tests and the existing tests that test partial downloads. I could split it, but it would duplicate the rest of the code.